fluss-client/src/test/java/com/alibaba/fluss/client/write/RecordAccumulatorTest.java

```diff
@@ -391,6 +392,41 @@ void testAwaitFlushComplete() throws Exception {
         assertThatThrownBy(accum::awaitFlushCompletion).isInstanceOf(InterruptedException.class);
     }
 
+    @Test
+    void testReadyWithTimeOut() throws Exception {
+        // test case: node1(tb1, tb2), node2(tb3).
+        IndexedRow row = row(DATA1_ROW_TYPE, new Object[] {1, "a"});
+        long batchSize = getTestBatchSize(row);
+        RecordAccumulator accum = createTestRecordAccumulator((int) batchSize, batchSize * 2);
+
+        // add bucket into cluster.
+        cluster = updateCluster(Arrays.asList(bucket1, bucket2, bucket3, bucket4));
+
+        // wait for ready timeout.
+        CompletableFuture<Void> future =
+                CompletableFuture.runAsync(
+                        () -> {
+                            accum.ready(cluster, 500L);
+                        });
+        assertThat(future).isNotCompleted();
+        Thread.sleep(1000L);
```

Comment:
Use `future.get()` or `future.join()` to wait async result. Never use sleep as it is not guaranteed to finish in the sleep time in different testing environment, and makes the tests unstable.